airsonic-main/src/main/java/org/airsonic/player/service/jukebox/AudioPlayer.java

```diff
@@ -49,16 +52,68 @@
 
     private final InputStream in;
     private final Listener listener;
-    private final SourceDataLine line;
     private final AtomicReference<State> state = new AtomicReference<State>(PAUSED);
+    private SourceDataLine line;
+    private AudioFormat format;
+    private DataLine.Info info;
     private FloatControl gainControl;
 
-    public AudioPlayer(InputStream in, Listener listener) throws Exception {
+    public AudioPlayer(InputStream in, String command, Listener listener) throws Exception {
         this.in = in;
         this.listener = listener;
 
-        AudioFormat format = new AudioFormat(AudioFormat.Encoding.PCM_SIGNED, 44100.0F, 16, 2, 4, 44100.0F, true);
-        line = AudioSystem.getSourceDataLine(format);
+        // SampleRate
+        String sampleRate = "";
+        Pattern r1 = Pattern.compile("\\-ar (.*?)\\ ");
+        Matcher sampleRate1 = r1.matcher(command);
+        while (sampleRate1.find()) {
+            sampleRate = sampleRate1.group(1);
+        }
+        LOG.debug("SampleRate: " + sampleRate);
+
+        // Channel
+        String channel = "";
+        Pattern r2 = Pattern.compile("\\-ac (.*?)\\ ");
+        Matcher channel2 = r2.matcher(command);
+        while (channel2.find()) {
+            channel = channel2.group(1);
+        }
+        LOG.debug("Channel: " + channel);
+
+        // Format
+        String pt = "";
+        Pattern r3 = Pattern.compile("\\-f (.*?)\\ ");
+        Matcher pt3 = r3.matcher(command);
+        while (pt3.find()) {
+            pt = pt3.group(1);
+        }
+        LOG.debug("Format: " + pt);
+
+        // Bits
+        String bits = pt.substring(1, 3);
+        LOG.debug("Bits: " + bits);
+
+        // Big/Little Endian
+        Boolean sl = true;
+        if (pt.substring(3, 5).equals("be")) {
+            sl = true;
+        }
```

Comment:
this condition not needed.

only need condition when variable changes (starts as true)